source/game/system/CourseMap.cc

```diff
@@ -233,6 +233,372 @@ CourseMap::~CourseMap() {
+    // The player is fully inside the current checkpoint, so just set to current checkpoint
+    case MapdataCheckPoint::SectorOccupancy::InsideSector:
+        id = checkpoint->id();
+        break;
+
+    // The player is between the sides of the quad, but NOT between this checkpoint and
+    // next; player is likely in the same checkpoint group
+    case MapdataCheckPoint::SectorOccupancy::OutsideSector_BetweenSides:
+        id = findSectorBetweenSides(pos, checkpoint, distanceRatio);
+        break;
+
+    // The player is not between the sides of the quad (may still be between this checkpoint
+    // and next); player is likely in a different checkpoint group
+    case MapdataCheckPoint::SectorOccupancy::OutsideSector:
+        id = findSectorOutsideSector(pos, checkpoint, distanceRatio);
+        break;
+
+    default:
+        break;
+    }
+
+    return id > -1 ? id : findSectorRegional(pos, checkpoint, distanceRatio);
+}
+
+s16 CourseMap::findSectorBetweenSides(const EGG::Vector3f &pos, MapdataCheckPoint *checkpoint,
+        f32 &distanceRatio) {
+    s16 id = -1;
+
+    // Behavior branches based on whether the player is closer to the next or previous checkpoint.
+    if (distanceRatio > 0.5f) {
+        // Step 1: Starting at current checkpoint, search forwards
+        id = searchNextCheckpoint(pos, 0, checkpoint, distanceRatio, false, false);
+
+        if (id != -1) {
+            return id;
+        }
+
+        // Step 2: If step 1 fails, start at next checkpoint(s) and search backwards
+        for (size_t i = 0; i < checkpoint->nextCount(); ++i) {
+            MapdataCheckPoint *next = checkpoint->nextPoint(i);
+
+            for (size_t j = 0; j < next->prevCount(); ++j) {
+                MapdataCheckPoint *prev = next->prevPoint(j);
+
+                if (prev == checkpoint) {
+                    continue;
+                }
+
+                id = findRecursiveSector(pos, 1, true, prev, distanceRatio, false);
+                if (id != -1) {
+                    return id;
+                }
+            }
+        }
+
+        // Step 3: If step 2 fails, start at previous checkpoint(s) and search forwards
+        for (size_t i = 0; i < checkpoint->prevCount(); ++i) {
+            MapdataCheckPoint *prev = checkpoint->prevPoint(i);
+
+            for (size_t j = 0; j < prev->nextCount(); ++j) {
+                MapdataCheckPoint *next = prev->nextPoint(j);
+
+                if (next == checkpoint) {
+                    continue;
+                }
+
+                id = findRecursiveSector(pos, 1, false, next, distanceRatio, false);
+                if (id != -1) {
+                    return id;
+                }
+            }
+        }
+
+        // Step 4: If step 3 fails, start at current checkpoint and search backwards
+        return searchPrevCheckpoint(pos, 0, checkpoint, distanceRatio, false, false);
+    } else {
+        // (do all the steps from 2a in reverse)
+        // Search prev
```

Comment:
```suggestion
        // Step 1: Starting at current checkpoint, search backwards
```
The other comments following this should be consistent.